pennylane/plugins/expt_tensornet.py

```diff
@@ -246,6 +228,43 @@ def var(self, observable, wires, par):
 
         return self.ev(obs_nodes_for_squares, wires) - self.ev(obs_nodes, wires)**2
 
+    def sample(self, observable, wires, par):
+
+        if not isinstance(observable, list):
+            observable, wires, par = [observable], [wires], [par]
+
+        matrices = [self._get_operator_matrix(o, p) for o, p in zip(observable, par)]
+
+        decompositions = [spectral_decomposition(A) for A in matrices]
+        eigenvalues, projector_groups = list(zip(*decompositions))
+        eigenvalues = list(eigenvalues)
+
+        # Matching each projector with the wires it acts on
+        # while preserving the groupings
+        projectors_with_wires = [[(proj, wires[idx]) for proj in proj_group]
+                                 for idx, proj_group in enumerate(projector_groups)]
+
+        # The eigenvalue - projector maps are preserved as product() preserves
+        # the previous ordering by creating a lexicographic ordering
+        joint_outcomes = list(product(*eigenvalues))
+        projector_tensor_products = list(product(*projectors_with_wires))
+
+        joint_probabilities = [] #[(outcome, 0.0) for idx, outcome in enumerate(joint_outcomes)]
+
+        for projs in projector_tensor_products:
+            obs_nodes = []
+            obs_wires = []
+            for proj, wire in projs:
```

Comment:
probably `wires` is better here to make clear that some observables might be multi-wire

Reply:
That's a good idea! Went with ``proj_wires`` so that it does not shadow the ``wires`` object from outer scope (and possibly cause confusion)